### ISSUE TITLE
Support `show`, `describe`, `analyze` lint for Trino

### ISSUE THREAD

**Issue by hongbo-miao** [sqlfluff/sqlfluff#5151]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

I have a file **trino_query.sql**:

```sql
show functions;

describe delta.hm_iot_db.motor;

analyze delta.hm_iot_db.motor;

show stats for delta.hm_iot_db.motor;
show stats for (select * from delta.hm_iot_db.motor where voltage > 10);
```

`sqlfluff lint --dialect=trino trino_query.sql` gives me results

```shell
== [trino_query.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'show
                       | functions'
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'describe
                       | delta.hm_iot_db.motor'
L:   5 | P:   1 |  PRS | Line 5, Position 1: Found unparsable section: 'analyze
                       | delta.hm_iot_db.motor'
L:   7 | P:   1 |  PRS | Line 7, Position 1: Found unparsable section: 'show
                       | stats for delta.hm_iot_db.motor'
L:   8 | P:   1 |  PRS | Line 8, Position 1: Found unparsable section: 'show
                       | stats for (select * from delta.hm_i...'
WARNING: Parsing errors found and dialect is set to 'trino'. Have you configured your dialect correctly?
```

It would be great to support `show`, `describe`, `analyze` lint for Trino in future. Thanks! 😃

### Dialect

trino

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
